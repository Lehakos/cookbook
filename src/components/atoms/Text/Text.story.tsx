import React from 'react';
import { storiesOf } from '@storybook/react';
import { select } from '@storybook/addon-knobs';

import Text from './Text';

const stories = storiesOf('Atoms|Text', module);

stories
  .add('No props', () => <Text>Some text</Text>)
  .add('Color prop', () => (
    <Text
      color={select(
        'color',
        ['primary', 'secondary', 'contrast', 'text'],
        'primary',
      )}
    >
      Some text
    </Text>
  ))
  .add('FontSize prop', () => (
    <Text fontSize={select('fontSize', [0, 1, 2, 3, 4], 3)}>Some text</Text>
  ))
  .add('FontWeight prop', () => (
    <Text fontWeight={select('fontWeight', [0, 1, 2], 2)}>Some text</Text>
  ));
