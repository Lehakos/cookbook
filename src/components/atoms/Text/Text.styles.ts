import styled from 'styled-components';

import { typography, TypographyProps } from 'shared/styledProps';

type WrapperProps = TypographyProps<HTMLDivElement>;

export const Wrapper = styled.div<WrapperProps>`
  ${typography}
`;

// eslint-disable-next-line no-undef
export { WrapperProps };
