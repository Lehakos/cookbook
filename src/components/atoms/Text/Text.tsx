import React from 'react';

import * as s from './Text.styles';

type Props = s.WrapperProps & {};

const Text: React.FunctionComponent<Props> = function ({ children, ...rest }) {
  return <s.Wrapper {...rest}>{children}</s.Wrapper>;
};

export default Text;
