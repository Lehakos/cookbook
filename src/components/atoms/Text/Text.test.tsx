import React from 'react';
import { render } from 'react-testing-library';

import Text from './Text';

describe('<Text />', () => {
  it('правильно отображается', () => {
    const { container } = render(<Text />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
