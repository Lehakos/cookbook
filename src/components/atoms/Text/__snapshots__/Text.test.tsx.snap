// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> правильно отображается 1`] = `
<div
  class=""
/>
`;
